src/magneticField/MagneticField.cpp

```diff
@@ -81,13 +81,35 @@ Vector3d MagneticFieldEvolution::getField(const Vector3d &position,
 Vector3d MagneticDipoleField::getField(const Vector3d &position) const {
 		Vector3d r = (position - origin);
 		Vector3d unit_r = r.getUnitVector();
-		
+
+
 		if (r.getR() == 0) { // singularity
 			return moment * 2 * mu0 / 3;
 		}
 		return (unit_r * (unit_r.dot(moment)) * 3 - moment) / pow(r.getR() / radius, 3) * mu0 / (4*M_PI);
 }
 
+Vector3d SingleModeHelicalMagneticField::getField(const Vector3d &position) const {
+		Vector3d r = (position - origin);
+
+		Vector3d e1 = unitVectorOrigin; // first of the unit vectors which span the polarization plane
+		if ((e1.x != 0.0) && (e1.y != 0.0) && (e1.z != 0.0)) {
+			e1 /= e1.getR();
```

Comment:
I do not undertand this - do you mean to prevent division by zero? If yes use `if e1.getR() > 0`, otherwise please add comment to explain this line.